src/main/java/com/couchbase/connect/kafka/sink/SubDocumentWriter.java

```diff
@@ -96,17 +98,29 @@ public Completable write(final AsyncBucket bucket, final JsonBinaryDocument docu
             }
         }
 
-        return mutation
-                .execute(persistTo, replicateTo)
-                .doOnError(new Action1<Throwable>() {
-                    @Override
-                    public void call(Throwable throwable) {
-                        if(createDocuments && throwable instanceof DocumentDoesNotExistException) {
-                            bucket.insert(JsonDocument.create(document.id())).toBlocking().single();
```

Comment:
indeed this is blocking in a callback and an issue.

Reply:
this is the code i was replacing with the retrybuilder, is it still somewhere in the files from the pull request?